src/python.rs

```diff
@@ -121,3 +151,54 @@ fn map_json_error(data: &[u8], json_error: JsonError) -> PyErr {
     let msg = format!("{} at {}", error_type, position);
     PyValueError::new_err(msg)
 }
+
+trait StringMaybeCache<'py> {
+    fn new(py: Python<'py>) -> Self;
+
+    fn get(&mut self, py: Python, json_str: &str) -> PyObject;
+}
+
+static STRINGS_CACHE: GILOnceCell<GILProtected<RefCell<AHashMap<String, PyObject>>>> = GILOnceCell::new();
+
+struct StringCache<'py>(RefMut<'py, AHashMap<String, PyObject>>);
+
+impl<'py> StringMaybeCache<'py> for StringCache<'py> {
+    fn new(py: Python<'py>) -> Self {
+        let protected_cache = STRINGS_CACHE.get_or_init(py, || GILProtected::new(RefCell::new(AHashMap::new())));
+        let strings_cache: RefMut<AHashMap<String, PyObject>> = protected_cache.get(py).borrow_mut();
+        Self(strings_cache)
```

Comment:
Taking a long borrow on a `RefMut` like this is not thread-safe, you'll end up causing other threads to panic if multiple threads are trying to do `jiter` deserialization in parallel. Which I think is definitely possible even with the GIL (e.g. it might switch threads in a Python validator function).

Either you need to make `STRINGS_CACHE` thread local or you need to make the refcell borrows short-lived just during reads / writes to the cache.

Reply:
how do I make `STRINGS_CACHE` thread local?

Reply:
Pushed a commit which makes the borrows short-lived.